wasp/wasp.py

```diff
@@ -118,6 +119,8 @@ def __init__(self):
         self.musicstate = {}
         self.musicinfo = {}
 
+        self._theme = array.array('H', b'\xef{\xef{\xef{\xef{\x00\xf8\xe0\x07<\xe7iJiJ\xef{\xb6\xb5\xb6\xbd\xff\xff')
+
```

Comment:
To be honest I was expecting direct lookup from the bytestring (e.g. encode the colour as one byte and reuse the _clut8_rgb565() functions (I think these are currently best documented in tools/rle_encode.py tool).

For spot colours I would expect the 256 colour palette to be sufficient.

Reply:
I didn't really understand this, could you elaborate a bit?

Reply:
I changed the theme storage to a `bytes` object (aka a `b'string'`).
This change will conserve a tiny bit of ram and will make it easier to set the theme.

Reply:
I'm not really sure what the benefits of using the CLUT here is.
We'd only save a few bytes at the cost of losing a lot of color options

I also think using the CLUT would make it harder for users to theme the watch by themselves - there are plenty of rgb565 color choosers online, but I'm not aware of any one of them that  supports wasp-os's CLUT.

Reply:
See attached `script` session:

https://hastebin.com/perahimete.rb

Reply:
> I'm not really sure what the benefits of using the CLUT here is.
> We'd only save a few bytes at the cost of losing a lot of color options

Converting the bytestring into an array makes a copy of the underlying data. Since the bytestring can be stored in flash it means we are comparing against a RAM overhead very close to zero.

Whether the theme uses a 65536 colour palette or a 256 colour palette doesn't matter nearly as much ensuring that wasp.system._theme is not format converted before being stored in the system class. If it started in flash in must stay in flash.


> I also think using the CLUT would make it harder for users to theme
> the watch by themselves - there are plenty of rgb565 color choosers
> online, but I'm not aware of any one of them that supports wasp-os's CLUT.

The 2-bit RLE encoder contains a function to lookup the nearest CLUT entry from a given RGB values. That can be used in the themer.

Note also that online colour choosers are pretty much useless for PineTime... the colour response is totally different to that of a typical RGB monitor. This is why there is a colour chooser in the selftest application... it's more or less impossible to design (attractive) colour applications for pinetime without using the built-in colour chooser.